lib/scrabble.rb

```diff
@@ -1,7 +1,24 @@
 class Scrabble
 
   def score(word)
-    1
+    return 0 if word == nil
+    point = 0
+    word.upcase.each_char do |letter|
+      point += point_values[letter]
+    end
+    point
+  end
+
+  def score_with_multipliers(word, multipliers, extra=1)
+    scored = word.upcase.chars.map.with_index do |letter, index|
+      multipliers.map.with_index do |number, num_index|
+        if num_index == index
+          point_values[letter] * number
+        end
+      end.compact
```

Comment:
Glad you got this to work. Don't think you need the nested `#map.with_index`. If you have the index at the outside level, you should just be able to call `multipliers[index]` to get the multiplier at the index that matches the index of the current letter.

This will also keep you from needing to use `#compact`.